scripts/test.mk

```diff
@@ -0,0 +1,52 @@
+# This file is part of the repository template project. Please, DO NOT edit this file.
+
+# The test types listed here are both those which might run both locally and in CI, or
+# in one but not the other.  All of the test types listed at
+# https://github.com/NHSDigital/software-engineering-quality-framework/blob/main/quality-checks.md
+# should be represented here with the exception of:
+#   - dependency scanning, which we expect to be applied at the repository level
+#   - secret scanning, which we expect to be a pre-commit hook
+#   - code review, which is outside the scope of automated testing for the moment
+
+test-integration:
+	@make _test name="integration"
+
+test-contract:
+	@make _test name="contract"
+
+test-ui:
+	@make _test name="ui"
+
+test-accessibility:
+	@make _test name="accessibility"
+
+test-lint:
+	@make _test name="lint"
+
+# test-code-quality covers checking for duplicate code, code smells, and dead code.
+test-code-quality:
+	@make _test name="code-quality"
+
+test-unit:
+	@make _test name="unit"
+
+test-coverage:
+	@make -test name="coverage"
+
+test: test-unit \
+	test-lint \
+	test-code-quality \
+	test-coverage \
+	test-contract \
+	test-ui \
+	test-integration \
+	test-accessibility
+
+_test:
+	@set -e ;\
```

Comment:
I put it there because I wanted a check on the specific line of shell that the make target executes, rather than the contents of the shell script.  Arguably I should also have a `set -u` to cover messing up the variable quoting, but putting it into the individual shell scripts wouldn't do what I want there.

Reply:
I've also added the `set -e` and a `set -u` to `scripts/test/unit` which should have been there anyway.

Reply:
I'm going to resolve this conversation for the moment. We're not in `.POSIX` mode, so `-e` isn't set by default.  Also, depending on what we decide to do with `.ONESHELL` I may well end up editing this anyway.